Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -4195,7 +4195,13 @@ void FrameLoader::continueLoadAfterNavigationPolicy(const ResourceRequest& reque
         return;
     }
 
-    m_client->dispatchWillSubmitForm(*formState, WTFMove(completionHandler));
+    Ref form = formState->form();
+    auto* submitter = formState->submitter();
+    auto formSubmission = FormSubmission::create(form.get(), submitter, form->attributes(), nullptr, LockHistory::No, formState->formSubmissionTrigger());
+    auto requestURL = formSubmission->requestURL();
+    String method = FormSubmission::Attributes::methodString(formSubmission->method());
+
+    m_client->dispatchWillSubmitForm(*formState, WTFMove(requestURL), WTFMove(method), WTFMove(completionHandler));
```

Comment:
Since this is v3 of this delegate method to add more data, I am a little curious whether the longer term solution is to expose an API equivalent of FormSubmission similar to _WKFrameInfo. this would allow to add more single fields without further complicating the port-specific code.